pandas/core/indexes/range.py

```diff
@@ -174,7 +179,7 @@ def _data(self):
 
     @cache_readonly
     def _int64index(self):
-        return Int64Index(self._data, name=self.name, fastpath=True)
+        return Int64Index._simple_new(self._data, name=self.name)
```

Comment:
shallow_copy?  ditto below

Reply:
And actually, `_shallow_copy` has not the ability to specify (override) the `name` (which is not really following how `_shallow_copy` works for other Index classes), so can also not use it there.

Might be a bit buggy in `_shallow_copy` (it is also not used at all anywhere in the RangeIndex implementation, although maybe in inherited methods ..). 
But I want to keep it here just about deprecating the fastpath keyword, so let's leave that for another issue.

Reply:
why are u changing these constructors at all?

Reply:
I am not changing anything. `fastpath=True` means exactly the `_simple_new` what I replaced it with. 
This PR does *not* try to do a clean-up of usage of `_simple_new` by replacing it with the default constructor, it only deprecates the `fastpath` argument.